provisioning_templates/snippet/kickstart_ifcfg_generic_interface.erb

```diff
@@ -20,7 +20,15 @@ BOOTPROTO="<%= @dhcp ? 'dhcp' : 'none' -%>"
 <%=   "IPV6_AUTOCONF=no" %>
 <%=   "IPV6ADDR=#{@interface.ip6}" %>
 <%-   if !@subnet6.nil? && @subnet6.gateway.present? -%>
-<%=     "IPV6_DEFAULTGW=#{@subnet6.gateway}" %><%= '%$real' if @subnet6.gateway.match(/^fe80:/) %>
+<%-
+  ipv6_defaultgw = @subnet6.gateway
+  unless @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.major.to_i >= 8
+    if @subnet6.gateway.match(/^fe80:/)
+      ipv6_defaultgw += '%$real'
+    end
+  end
+-%>
+<%=     "IPV6_DEFAULTGW=#{ipv6_defaultgw}" %>
```

Comment:
This is not correct, this is undocumented but still supported: https://github.com/NetworkManager/NetworkManager/blob/master/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L2056-L2060

IPV6_DEFAULTDEV=dev is preferred but for backward compatibility I'd keep the old behavior.

Reply:
I'll link to [a specific commit](https://github.com/NetworkManager/NetworkManager/blob/235cb4a5d310e207d4751c651d3e672481aa30dd/src/settings/plugins/ifcfg-rh/nms-ifcfg-rh-reader.c#L2022-L2051), because the current link in your review already points to other code ...

As I understand it, the `IPV6_DEFAULTDEV` is used as fallback if there is no device specified with `%devXYZ`.

I'm not sure anymore why I suggested this change, but it must have been necessary for me - I wouldn't change something just because... I'll try to find the reason :-)